### DISCUSSION TITLE
Colocar um empacotador

### DISCUSSION THREAD

**Discussion by mateusmoutinho** [DesignLiquido/delegua#670] [Ideias]
Olá, curti muito  o projeto , e acho que tem um grande potencial para ensinar programação para iniciantes, 
porém notei uma falha no projeto de vocês, a nescessidade de ter o npm, o node pré instalado na máquina do usuário,
isso além de criar vários problemas de versões, já que o node é extreamente problemático para isso, quanto a nescessidade de obrigar um usário iniciante a ter essas dependencias instaladas.
Pra resolver isso , criei um protótipo (já funciona em linux) de uma versão completamente empacotada do delegua
essa versão não depende de internet (ideal para locais remotos), não depende do node, do npm , ou qualquer dependência.
segue abaixo o repo da versão empacotada: 

https://github.com/OUIsolutions/delegua_bin?tab=readme-ov-file

**Comment by ItaloCobains** [comment]
@leonelsanchesdasilva @cx-leonelsanches

**Comment by leonelsanchesdasilva** [comment]
@mateusmoutinho Parabéns pela iniciativa! Gostei bastante. 

Tenho algumas sugestões em como esse empacotamento pode melhorar, por exemplo:

- Instruções passo a passo sobre como instalar em ambientes como Windows, Linux e Mac. Como você mesmo disse, é para iniciantes, mas vou além e imagino que professores (sobretudo em países com escassez de internet, como Angola e Moçambique) se beneficiem desse instalador. 
- Instruções para outras pessoas atualizarem este repositório, como por exemplo detalhes sobre a tecnologia do instalador, quais arquivos fazem o que, e assim por diante.

**Comment by mateusmoutinho** [comment]
Opa, brigadão man ,vamos aos tópicos:

instruções sobre o código fonte do instalador (escrito em C para poder rodar nativamentee e embedar as dependencias e o node internamente) , isso será incluido sim na versão final.

Instrruções de como usar o instalador, não acho isso nescessário,já que a versão final do instalador será gráfica, e não uma cli como é agora, então a ideia é ser igual o instalador do chrome, você só clica, next..next..done. 

Também pretendo segmentar em 3 repositórios diferentes (windows/linux/mac), assim controlamos melhor as versões do node, e os arquivos que vão em cada.

Devido a OUI não ser uma empresa voltado para  o universo de ensino, (apesar de termos bibliotecas e  frameworks open source que nos criamos e mantemos), nos atuamos no segmento de automação  jurídica, então, apesar de sermos open source friendly, e termos um compromisso forte com o código aberto, nosso core esta na prestação de serviço e em nossos erps. 

Por isso gostaria de passar a ownership dos instaladores pra vocẽs conforme eles fiquem pronto, apenas pedindo caso possível , que deixem as notas de rodapé que foi uma contribuição nossa.  
E também a gente gostaria de saber se o instalador (após se tornar production ready) , vai ser usado como um dos mecanismos oficiias da design líquido de instalação do delégua.

**Reply by leonelsanchesdasilva** [reply]
Legal! Então você pretende usar:

- Para Windows: https://www.firegiant.com/docs/wix/v3/tutorial/
- Para Linux: https://wiki.ubuntu.com/PbuilderHowto
- Para Mac: https://stackoverflow.com/questions/11487596/making-macos-installer-packages-which-are-developer-id-ready

Eu fico no aguardo do que vocês forem fazer, e vamos alegremente aceitar os repositórios quando os instaladores estiverem prontos. Muito show a contribuição de vocês.

Nós não apenas iremos colocar notas de rodapé como vamos citá-los como nossos parceiros, tanto no nosso site quando na página da organização do GitHub, e já agradecemos antecipadamente pela contribuição.

**Comment by mateusmoutinho** [comment]
Opa man , brigadão pela resposta, sobre a gui eu pretendo usar gtk por pura zona de conforto kkk ( pq eu ja usei gtk), mas vou ler as docs, e se for fácil vou usar essa que vocẽ recomendou , ai vou ver com o ítalo pra ainda essa semana , a gente ja estabilizar o modo cli , tanto linux quanto windows. depois eu jogo a gui em ambos.

**Reply by ItaloCobains** [reply]
up
